File: Ticketa.Core/DTOs/BookingResultDto.cs
namespace Ticketa.Core.DTOs
{
  public class BookingResultDto
  {
    public bool Succeeded { get; set; }
    public bool IsNotFound { get; set; }
    public string? BookingReference { get; set; }
    public decimal? TotalAmount { get; set; }
    public List<SeatDto> ConflictingSeats { get; set; } = [];

    public static BookingResultDto Success(string refrence, decimal total) => new()
    {
      Succeeded = true,
      BookingReference = refrence,
      TotalAmount = total
    };

    public static BookingResultDto NotFound() => new() { IsNotFound = true };

    public static BookingResultDto Conflict(IEnumerable<SeatDto> seats) => new()
    {
      Succeeded = false,
      ConflictingSeats = seats.ToList()
    };
  }
}

File: apps/mobile/lib/features/booking/data/models/booking_result_dto.dart
import 'seat_dto.dart';

class BookingResultDto {
  final bool succeeded;
  final String? bookingReference;
  final double? totalAmount;
  final List<SeatDto> conflictingSeats;

  const BookingResultDto({
    required this.succeeded,
    this.bookingReference,
    this.totalAmount,
    this.conflictingSeats = const [],
  });

  factory BookingResultDto.fromJson(Map<String, dynamic> json) => BookingResultDto(
        succeeded: true,
        bookingReference: json['bookingReference']?.toString(),
        totalAmount: (json['totalAmount'] as num?)?.toDouble(),
      );

  factory BookingResultDto.conflict(Map<String, dynamic> json) {
    final seats = (json['conflictingSeats'] as List<dynamic>?)
            ?.map((e) => SeatDto.fromJson(e as Map<String, dynamic>))
            .toList() ??
        [];
    return BookingResultDto(
      succeeded: false,
      conflictingSeats: seats,
    );
  }
}
